### PR TITLE
PLANNER-2658 Register services unconditionally

### PR DESCRIPTION
**Pull request by triceo**
Signed-off-by: Lukáš Petrovický <lpetrovi@redhat.com>

### Checklist
- [ ] Documentation updated if applicable.
- [ ] Upgrade recipe provided if applicable.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>specific pull request build</b> please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] tests</b>
* for a <b>full downstream build</b> 
  * for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  * for <b>github actions</b> job: add the label `run_fdb`
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
* for a <b>specific Quarkus LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] LTS</b>
* for a <b>Native check</b> please add comment: <b>Jenkins run native</b>
* for a <b>specific Native LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] native</b>
</details>

### CODE REVIEW

**Comment by kie-ci1** [comment]
**(tests) - optaplanner-quickstarts job** `#358` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/optaplanner.tests.downstream.optaplanner-quickstarts/358/display/redirect or see console log:
<details>
<summary><b>Console Logs</b></summary>
[2022-02-17T08:01:21.216Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1645084876708/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/esm/c_data_list__item_m_expanded_m_selectable_before_BackgroundColor.js’: No such file or directory<br/>[2022-02-17T08:01:21.216Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1645084876708/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/esm/c_data_list__item_m_selectable_active_BoxShadow.js’: No such file or directory<br/>[2022-02-17T08:01:21.216Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1645084876708/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/esm/c_data_list__item_m_selectable_focus_BoxShadow.js’: No such file or directory<br/>[2022-02-17T08:01:21.216Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1645084876708/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/esm/c_data_list__item_m_selectable_hover_BoxShadow.js’: No such file or directory<br/>[2022-02-17T08:01:21.216Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1645084876708/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/esm/c_data_list__item_m_selectable_hover_ZIndex.js’: No such file or directory<br/>[2022-02-17T08:01:21.217Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1645084876708/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/esm/c_data_list__item_m_selectable_OutlineOffset.js’: No such file or directory<br/>[2022-02-17T08:01:21.217Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1645084876708/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/esm/c_data_list__item_m_selected_before_BackgroundColor.js’: No such file or directory<br/>[2022-02-17T08:01:21.217Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1645084876708/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/esm/c_data_list__item_m_selected_BoxShadow.js’: No such file or directory<br/>[2022-02-17T08:01:21.217Z] 656M	/home/jenkins/workspace<br/>[2022-02-17T08:01:21.217Z] 0	/home/jenkins/.pki<br/>[2022-02-17T08:01:21.217Z] 118M	/home/jenkins/tools<br/>[2022-02-17T08:01:21.217Z] 0	/home/jenkins/caches<br/>[2022-02-17T08:01:21.217Z] 144M	/home/jenkins/.npm<br/>[2022-02-17T08:01:21.217Z] 3.1G	/home/jenkins<br/>[Pipeline] }<br/>[Pipeline] // dir<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Install build-chain tool)<br/>Stage "Install build-chain tool" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Build projects)<br/>Stage "Build projects" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Sonar analysis)<br/>Stage "Sonar analysis" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (check space after build)<br/>Stage "check space after build" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>[2022-02-17T08:01:22.104Z] + wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/optaplanner.tests.downstream.optaplanner-quickstarts/358/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

**Comment by kie-ci1** [comment]
**(tests) - optaplanner job** `#357` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/optaplanner.tests.optaplanner/357/display/redirect or see console log:
<details>
<summary><b>Console Logs</b></summary>
[2022-02-17T08:08:48.290Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1645085323783/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/esm/c_button_m_control_hover_after_BorderBottomColor.js’: No such file or directory<br/>[2022-02-17T08:08:48.290Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1645085323783/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/esm/c_button_m_control_hover_after_BorderBottomWidth.js’: No such file or directory<br/>[2022-02-17T08:08:48.290Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1645085323783/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/esm/c_button_m_control_m_expanded_after_BorderBottomColor.js’: No such file or directory<br/>[2022-02-17T08:08:48.290Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1645085323783/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/esm/c_button_m_control_m_expanded_after_BorderBottomWidth.js’: No such file or directory<br/>[2022-02-17T08:08:48.290Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1645085323783/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/esm/c_button_m_danger_active_BackgroundColor.js’: No such file or directory<br/>[2022-02-17T08:08:48.290Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1645085323783/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/esm/c_button_m_danger_active_Color.js’: No such file or directory<br/>[2022-02-17T08:08:48.290Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1645085323783/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/esm/c_button_m_danger_BackgroundColor.js’: No such file or directory<br/>[2022-02-17T08:08:48.290Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1645085323783/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/esm/c_button_m_danger_Color.js’: No such file or directory<br/>[2022-02-17T08:08:48.290Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1645085323783/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/esm/c_button_m_danger_focus_BackgroundColor.js’: No such file or directory<br/>[2022-02-17T08:08:48.290Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaweb-vehicle-routing_ws-cleanup_1645085323783/bc/kiegroup_optaweb_vehicle_routing/optaweb-vehicle-routing/optaweb-vehicle-routing-frontend/node_modules/@patternfly/react-tokens/dist/esm/c_button_m_danger_focus_Color.js’: No such file or directory<br/>[2022-02-17T08:08:48.290Z] 654M	/home/jenkins/workspace/KIE/kogito/main<br/>[2022-02-17T08:08:48.290Z] 654M	/home/jenkins/workspace/KIE/kogito<br/>[2022-02-17T08:08:48.290Z] 654M	/home/jenkins/workspace/KIE<br/>[2022-02-17T08:08:48.290Z] 654M	/home/jenkins/workspace<br/>[Pipeline] }<br/>[Pipeline] // dir<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Install build-chain tool)<br/>Stage "Install build-chain tool" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Build projects)<br/>Stage "Build projects" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Sonar analysis)<br/>Stage "Sonar analysis" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (check space after build)<br/>Stage "check space after build" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>[2022-02-17T08:08:49.173Z] + wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/optaplanner.tests.optaplanner/357/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

**Comment by triceo** [comment]
Jenkins rerun optaplanner tests

File: optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java [hunk 1]
```diff
@@ -108,8 +109,9 @@ FeatureBuildItem feature() {
     }
 
     @BuildStep
-    void registerScoreDirectorFactorySpi(BuildProducer<ServiceProviderBuildItem> services) {
-        registerSpi(ScoreDirectorFactoryService.class, services);
+    void registerSpi(BuildProducer<ServiceProviderBuildItem> services) {
+        Stream.of(ScoreDirectorFactoryService.class, JoinerService.class)
+                .forEach(service -> registerSpi(service, services));
```
Comment (yurloc) [inline optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java:114]: (nitpick) Stream seems like an overkill here. You could simply
```java
registerSpi(ScoreDirectorFactoryService.class, services);
registerSpi(JoinerService.class, services);
```

**Review by yurloc — APPROVED** [review]
LGTM.

File: optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java (hunk 1 above)
Comment (triceo) [reply optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java:114]: I see your point. In the interest of fixing the build quickly, I will keep it as is, but I will remember it for when I surely touch this code in the future.

Comment (yurloc) [reply optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java:114]: No big deal :slightly_smiling_face:

**Pull request merged**